R/geom-.r

```diff
@@ -106,12 +106,19 @@ Geom <- ggproto("Geom",
 
   setup_data = function(data, params) data,
 
+  # evaluate defaults according to theme
+  eval_defaults = function(self, theme) {
+    if (length(theme) == 0) theme <- theme_grey()
+
+    lapply(self$default_aes, rlang::eval_tidy, data = list(theme = theme))
```

Comment:
This seems like the right approach to me, but I think I'd embed I do:

```R
from_theme <- function(name) {
  # plus error handling
  theme$geom[[name]]
}
```

Then `data = list(from_theme = from_theme)`